### PR TITLE
Authenticated registry tests

### PR DESCRIPTION
**Pull request by stevenhorsman**
Add the three new test scenarios described in the issue for containerd w/ crictl and kubernetes 

Fixes: #5209
Depends-on: github.com/kata-containers/kata-containers#5680

### CODE REVIEW

**Comment by stevenhorsman** [comment]
I've added in the secret binding to pass through the registry credentials to the  REGISTRY_CREDENTIAL_ENCODED envvar  starting with the http://jenkins.katacontainers.io/view/CCv0/job/tests-CCv0-ubuntu-20.04-x86_64-CC_CRI_CONTAINERD-PR/ and http://jenkins.katacontainers.io/view/CCv0/job/kata-containers-CCv0-ubuntu-20.04-x86_64-CC_CRI_CONTAINERD-PR, so now I'll try /test-ubuntu

**Comment by stevenhorsman** [comment]
I might need a second pair of eyes on this. The k8s tests are passing but the crictl based ones failing when trying to exec into the container as `crictl ps` shows no containers:
```
containers: CONTAINER           IMAGE               CREATED             STATE               NAME                ATTEMPT             POD ID, container_id:
```
but the log shows that the pull image worked:
```
"{\"msg\":\"pull and unpack image \\\"confidential-containers-auth_nginx\\\", with image-rs succeeded \",\"level\":\"INFO\",\"ts\":\"2022-11-26T11:50:37.877630933Z\",\"source\":\"agent\",\"name\":\"kata-agent\",\"subsystem\":\"rpc\",\"pid\":\"119\",\"version\":\"0.1.0\"}"
```
I'll retry it on Monday to check that my recently changes didn't break anything

**Comment by stevenhorsman** [comment]
I've set up the credentials binding for the all the kata-container's Jenkins hosted jobs, and all the expected ones are passing, so I think that all that is left is to work with @fidencio, or @arronwy to set the correct config for TDX jobs and hopefully this can be merged 🤞

**Comment by fidencio** [comment]
> I've set up the credentials binding for the all the kata-container's Jenkins hosted jobs, and all the expected ones are passing, so I think that all that is left is to work with @fidencio, or @arronwy to set the correct config for TDX jobs and hopefully this can be merged crossed_fingers

I've done the configuration on the TDX Jenkins side, let's re-reun the jobs and see if they pass as expected.

**Comment by stevenhorsman** [comment]
Unfortunately due to the issues we had with s390x support for the ring package, which led us to pin the s390x version of image-rs to an old version in https://github.com/kata-containers/kata-containers/pull/5600, the new code won't compile for s390x until https://github.com/confidential-containers/image-rs/issues/80 is resolved (and it seems to be making good progress). I think we need to decide if this is a blocker or not for this PR. @BbolroC - do you have any thoughts on this?

**Comment by BbolroC** [comment]
@stevenhorsman thanks for sharing. I think https://github.com/confidential-containers/image-rs/issues/80 will be merged in the near future (including a new PR to revert https://github.com/kata-containers/kata-containers/pull/5600 and exclude a cosign for s390x. one question: do you guys open the PR or want me to do it?). We could ignore test results for s390x until then. :wink:

**Comment by stevenhorsman** [comment]
> @stevenhorsman thanks for sharing. I think [confidential-containers/image-rs#80](https://github.com/confidential-containers/image-rs/issues/80) will be merged in the near future (including a new PR to revert [kata-containers/kata-containers#5600](https://github.com/kata-containers/kata-containers/pull/5600) and exclude a cosign for s390x. one question: do you guys open the PR or want me to do it?). We could ignore test results for s390x until then. 😉

I think Matt will be creating the PR to bump image-rs and revert [kata-containers/kata-containers#5600](https://github.com/kata-containers/kata-containers/pull/5600) and after that I'm hoping this will work (it might need some work on the multi-arch test image, but I can try picking up that. If we are happy to merge this without s390x then if that happens I'll create a follow on task to get this working on s390x and take ownership of that. Thanks!

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

**Pull request merged**
